### ISSUE TITLE
A different fuzzer error

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/area-builder#73]
```
Error: Assertion failed: Error: Attempt to remove shape that is not a resident.
    at window.assertions.assertFunction (http://192.168.0.2:8080/assert/js/assert.js:22:13)
    at ShapePlacementBoard.inherit.removeResidentShape (http://192.168.0.2:8080/area-builder/js/common/model/ShapePlacementBoard.js?bust=1424765341887:294:45)
    at removalListener (http://192.168.0.2:8080/area-builder/js/common/model/ShapePlacementBoard.js?bust=1424765341887:926:14)
    at Array.wrapper (http://192.168.0.2:8080/axon/js/Property.js?bust=1424765341887:226:11)
    at Property.inherit._notifyObservers (http://192.168.0.2:8080/axon/js/Property.js?bust=1424765341887:105:29)
    at Property.inherit._setAndNotifyObservers (http://192.168.0.2:8080/axon/js/Property.js?bust=1424765341887:91:14)
    at Property.inherit.set (http://192.168.0.2:8080/axon/js/Property.js?bust=1424765341887:64:16)
    at MovableShape.Object.defineProperty.set [as userControlled] (http://192.168.0.2:8080/axon/js/PropertySet.js?bust=1424765341887:134:43)
    at Object.ShapeNode.addInputListener.SimpleDragHandler.start (http://192.168.0.2:8080/area-builder/js/common/view/ShapeNode.js?bust=1424765341887:144:37)
    at SimpleDragHandler.inherit.startDrag (http://192.168.0.2:8080/scenery/js/input/SimpleDragHandler.js?bust=1424765341887:151:22)
```

**Comment by jbphet** [comment]
After doing a bunch of tests, I think this may be due to a race condition where the eraser button is pressed and then a shape is grabbed just before it starts to fade out.  It looks like there is a one frame window where this would be possible.  I'll try making the fade proportion change as soon as fading away is started, which will trigger a listener that will make the shape unpickable, which should close this window.

**Comment by jbphet** [comment]
This has been fixed, and fuzz testing on two machines for 1.5 hrs on each resulted in no occurrences of this error or any others.  The test server also gives it a green light now.  Closing.
